test/torchaudio_unittest/functional/functional_impl.py

```diff
@@ -382,6 +382,31 @@ def test_phase_vocoder_shape(self, rate, test_pseudo_complex):
         output_shape = (torch.view_as_complex(spec_stretch) if test_pseudo_complex else spec_stretch).shape
         assert output_shape == expected_shape
 
+    @parameterized.expand(
+        [
+            ["abc", "", 3],
+            ["", "", 0],
+            ["abc", "abc", 0],
+            ["aaa", "aba", 1],
+            ["aba", "aaa", 1],
+            ["aa", "aaa", 1],
+            ["aaa", "aa", 1],
+            ["abc", "bcd", 2],
+        ]
+    )
```

Comment:
Can you organize the test cases;
1. Split test methods for `equal`, `deletion`, `substitution`.
1. For the case of `deletion` and `substitution` methods, check the swapped case inside of the method.
1. For the `deletion` and `substitution` methods, add cases where there are 1, 2 and 3 occurrences and make sure that they work even when the event occurs at the beginning and/or end of string.
1. For all the methods, please include the cases with empty string and string with multi-byte (unicode) characters, with some strings that are under [unicode equivalence](https://en.wikipedia.org/wiki/Unicode_equivalence).

Reply:
I've added these tests and grouped them. However, we may have problem for unicode with torchscript. Please see below.